src/libcore/ptr/mod.rs

```diff
@@ -248,17 +248,16 @@ pub(crate) struct FatPtr<T> {
 /// # Examples
 ///
 /// ```rust
-/// #![feature(slice_from_raw_parts)]
 /// use std::ptr;
 ///
 /// // create a slice pointer when starting out with a pointer to the first element
-/// let mut x = [5, 6, 7];
-/// let ptr = &mut x[0] as *mut _;
-/// let slice = ptr::slice_from_raw_parts_mut(ptr, 3);
+/// let x = [5, 6, 7];
+/// let ptr = &x[0] as *const _;
```

Comment:
Currently miri considers this (more precisely, the access to an element other than the first in the line below) UB. See https://github.com/rust-lang/unsafe-code-guidelines/issues/134 for details and discussion whether it should be that way. Regardless of the outcome of that discussion, it seems better to side-step the issue by using `x.as_ptr()` here.

Reply:
Yes, `as_ptr` is definitely the correct choice here. I didn't even notice this error in the docs; good catch! (In my defense, I only removed the `mut` bit of what was already written.)

Reply:
Good catch indeed! We really need to work on https://github.com/rust-lang/miri/issues/584...